### PR TITLE
Team stats

### PR DESCRIPTION
**Pull request by Ashkanthegreat**
Completed Team Stats methods.

### CODE REVIEW

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -96,30 +80,36 @@ def count_of_teams
     @teams.length
   end
 
-  def average_goals_by_team(team_id, hoa = nil)
-    goals = 0
-    games = 0
+  def average_goals_by_team(team_id, hoa = nil) # game_teams?
```
Comment (Ashkanthegreat) [inline lib/stat_tracker.rb:83]: Lets look at this to refactor.

File: lib/stat_tracker.rb [hunk 2]
```diff
@@ -152,10 +142,16 @@ def lowest_scoring_home_team
...
 
-  def most_accurate_team(season)
+  def team_games_by_season(season) # test
     season_games = @games.find_all{|game| game.season == season}
     season_game_ids = season_games.map{|game| game.game_id}
-    team_performances = @game_teams.find_all{|team| season_game_ids.include?(team.game_id)}
+    @game_teams.find_all{|team| season_game_ids.include?(team.game_id)}
+  end
+
+  # make a season class?
```
Comment (Ashkanthegreat) [inline lib/stat_tracker.rb:151]: Make the season class as a parent?

File: lib/stat_tracker.rb [hunk 3]
```diff
@@ -208,13 +196,92 @@ def most_tackles(season)
...
+  def worst_season(team_id)
+    team_seasons = @games.find_all {|game| game.home_team_id  == team_id || game.away_team_id == team_id }
+    seasons = team_seasons.group_by {|game| game.season}
+    seasons.each do |season, season_games|
+      season_game_ids = season_games.map{|game| game.game_id}
+      team_games = @game_teams.find_all { |game| game.team_id == team_id && season_game_ids.include?(game.game_id)}
+      seasons[season] = calculate_win_percentage(team_games)
+    end
+    seasons.min_by { |season, win_pct|
+      win_pct }[0]
```
Comment (jpc20) [inline lib/stat_tracker.rb:241]: Great job getting these to work guys. Let's refactor them into helper methods when we get the chance.

File: lib/game.rb [hunk 4]
```diff
@@ -28,8 +28,8 @@ def initialize(game_info)
     @season = game_info[:season]
     @type = game_info[:type]
     @date_time = game_info[:date_time]
-    @away_team_id = game_info[:away_team_id].to_i
-    @home_team_id = game_info[:home_team_id].to_i
+    @away_team_id = game_info[:away_team_id]
```
Comment (arpariseau) [inline lib/game.rb:31]: Spec harness wants team_id as a string, so had to fix it here too so that they match

File: test/stat_tracker_test.rb [hunk 5]
```diff
@@ -144,4 +156,46 @@ def test_it_can_find_team_with_most_season_tackles
...
+
+  def test_it_can_return_a_teams_info
+     assert_equal ({"team_id" => "1",
+       "franchise_id" => "23",
+       "team_name" => "Atlanta United",
+       "abbreviation" => "ATL",
+       "link" => "/api/v1/teams/1"}), @stat_tracker.team_info("1")
+  end
+
+  def test_it_can_find_teams_best_season
```
Comment (Ashkanthegreat) [inline test/stat_tracker_test.rb:168]: Lets look at making this test and the next test more robust.

File: lib/game_team.rb [hunk 6]
```diff
@@ -30,7 +30,7 @@ def self.all
               :takeaways
   def initialize(game_team_info)
     @game_id = game_team_info[:game_id].to_i
-    @team_id = game_team_info[:team_id].to_i
+    @team_id = game_team_info[:team_id]
```
Comment (arpariseau) [inline lib/game_team.rb:33]: Same as with game, changing team_id to match spec harness requirements

File: lib/stat_tracker.rb [hunk 7]
```diff
@@ -96,30 +80,36 @@ def count_of_teams
     @teams.length
   end
 
-  def average_goals_by_team(team_id, hoa = nil)
-    goals = 0
-    games = 0
+  def average_goals_by_team(team_id, hoa = nil) # game_teams?
+    goals = total_games_and_goals_by_team(team_id, hoa)[0]
```
Comment (jpc20) [inline lib/stat_tracker.rb:84]: Refactored this method to have the 2 helper methods below.
Used a mock and stub on the test.

File: lib/stat_tracker.rb [hunk 8]
```diff
@@ -96,30 +80,36 @@ def count_of_teams
...
-    return 0 if games == 0
-    (goals.to_f / games.to_f).round(2)
+    goals_games
   end
 
-  def unique_team_ids
-    @game_teams.uniq { |game_team| game_team.team_id }.map{ |game_team| game_team.team_id }
+  def add_goals_and_games(goals_games, game_team)
+    goals_games[0] += game_team.goals
+    goals_games[1] += 1
```
Comment (arpariseau) [inline lib/stat_tracker.rb:104]: if we're incrementing by goals each time, can just call games.length at the end instead

File: lib/stat_tracker.rb [hunk 9]
```diff
@@ -208,13 +196,92 @@ def most_tackles(season)
...
+  def fewest_goals_scored(team_id)
+    all_games_by_team(team_id).min_by{|game| game.goals}.goals
+  end
+
+  def favorite_opponent(team_id)
+    team_games = all_games_by_team(team_id)
+    team_games.map! {|game| game.game_id}
+    opp_team_games = @game_teams.find_all {|game| team_games.include?(game.game_id) && game.team_id != team_id}
+    opp_teams = opp_team_games.group_by {|game| game.team_id}
+    win_pct_against = opp_teams.transform_values{|game| 1 - calculate_win_percentage(game)}
```
Comment (arpariseau) [inline lib/stat_tracker.rb:268]: Have to use 1 - win percentage to calculate the percentage against

**Pull request merged**